src/clear/model/json_serializable.cr

```diff
@@ -0,0 +1,71 @@
+macro columns_to_instance_vars
+  struct Assigner
+    include JSON::Serializable
+
+    {% for name, settings in COLUMNS %}
+      getter {{name.id}} : {{settings[:type]}}?
+    {% end %}
+
+    def to_create
+      parse_from_json({{@type}}.new) # Loop through instance variables and assign to the newly created orm instance
+    end
+
+    def to_update(to_update_model)
+      parse_from_json(to_update_model) # Loop through instance variables and assign to the orm instance you are updating
+    end
+
+    macro finished
+      def parse_from_json(model)
+        {% for name, settings in COLUMNS %}
+          model.{{name.id}} = @{{name.id}}.not_nil! unless @{{name.id}}.nil?
+        {% end %}
+
+        model
+      end
+    end
+  end
+
+  def self.from_json(string_or_io : String | IO)
+    Assigner.from_json(string_or_io)
```

Comment:
```suggestion
    Assigner.from_json(string_or_io).create
```

Reply:
This method was already taken by

```
def self.new(string_or_io : String | IO)
    Assigner.from_json(string_or_io).create
end
```

The reason why I have a separate `self.from_json` method is because it is an intermediate JSON parsing stage that after parsing from, people can do something with it before `.create` or `.update`

Do you think this is a good idea?

Reply:
i don't think you want to return an `Assigner` here

Reply:
So I should just not have the method then?